mindmeld/stemmers.py

```diff
@@ -52,15 +62,56 @@ def stem_word(self, word):
         return word if stem == "" else stem
 
 
-class SpanishNLTKStemmer(Stemmer):
+class SnowballNLTKStemmer(Stemmer):
+
     @property
     def _stemmer(self):
         # lazy init the stemmer
         if not hasattr(self, "__stemmer"):
-            setattr(self, "__stemmer", nltk.stem.SnowballStemmer("spanish"))
+            setattr(self, "__stemmer", nltk.stem.SnowballStemmer(self.language))
         return getattr(self, "__stemmer")
 
     def stem_word(self, word):
         stem = word.lower()
         stem = self._stemmer.stem(stem)
         return word if stem == "" else stem
+
+
+class NoOpStemmer(Stemmer):
+
+    @property
+    def _stemmer(self):
+        return
+
+    def stem_word(self, word):
+        return word
+
+
+def get_language_stemmer(language_code):
+
+    if not language_code:
+        return NoOpStemmer()
+
+    language_code = language_code.lower()
+
+    if language_code == 'en':
+        return EnglishNLTKStemmer()
+
+    language = None
+    if len(language_code) == 2:
+        language = pycountry.languages.get(alpha_2=language_code)
+    elif len(language_code) == 3:
+        language = pycountry.languages.get(alpha_3=language_code)
+
+    if not language:
+        logger.warning('Language code %s is not supported for stemming. '
+                       'Consider disabling the stemmer in config.py.', language_code)
+        return NoOpStemmer()
+
+    language_name = language.name.lower()
+    if language_name in nltk.stem.SnowballStemmer.languages:
+        return SnowballNLTKStemmer(language_name)
+
+    logger.warning('Language code %s is not supported for stemming. '
+                   'Consider disabling the stemmer in config.py.', language_code)
+    return NoOpStemmer()
```

Comment:
This is great. Is disabling the stemmer a different result than getting the NoOpStemmer?

Reply:
This is a good point. So Mindmeld stems all queries but uses the stemmed word as a feature only when stemming is enabled. I will make this clear in the warning message.